superset/views/core.py

```diff
@@ -1796,46 +1795,29 @@ def testconn(self):
             uri = request.json.get('uri')
             db_name = request.json.get('name')
             impersonate_user = request.json.get('impersonate_user')
-            database = None
+
+            # extras is sent as json, but required to be a string in the Database model
+            extra_json = request.json.get('extras', {})
+            extra_str = json.dumps(extra_json)
+
             if db_name:
-                database = (
+                existing_database = (
```

Comment:
Would you be able to clarify the difference between `existing_database` and `database` (defined below)?